web/yo/app/scripts/controllers/gene.js

```diff
@@ -24,63 +24,107 @@ angular.module('oncokbApp')
 
             $scope.addMutation = function(newMutationName) {
                 if (this.gene && newMutationName) {
-                    newMutationName = newMutationName.toString().trim();
-                    var exists = false;
-                    var removed = false;
-                    var tempMutation;
-                    var isVUS = false;
-                    var mutationNameBlackList = [
-                        'activating mutations',
-                        'activating mutation',
-                        'inactivating mutations',
-                        'inactivating mutation'
-                    ];
-                    this.gene.mutations.asArray().forEach(function(e) {
+                    if ($scope.validateMutation(newMutationName, true) === true) {
+                        $scope.realtimeDocument.getModel().beginCompoundOperation();
+                        var _mutation = $scope.realtimeDocument.getModel().create(OncoKB.Mutation);
+                        var filteredContent = [];
+                        _.each(newMutationName.split(','), function(item) {
+                            item = item.trim();
+                            if (item.length > 0) {
+                                filteredContent.push(item);
+                            }
+                        });
+                        _mutation.name.setText(filteredContent.join(','));
+                        _mutation.oncogenic_eStatus.set('obsolete', 'false');
+
+                        this.gene.mutations.push(_mutation);
+                        $scope.realtimeDocument.getModel().endCompoundOperation();
+                        $scope.initGeneStatus(_mutation);
+                    }
+                }
+            };
+            $scope.validateMutation = function(newMutationName, alert, firstEnter, mutation) {
+                var exists = false;
+                var removed = false;
+                var tempMutation;
+                var isVUS = false;
+                var duplication = false;
+                var mutationNameBlackList = [
+                    'activating mutations',
+                    'activating mutation',
+                    'inactivating mutations',
+                    'inactivating mutation'
+                ];
+                if (firstEnter) {
+                    var mutationIndex = $scope.gene.mutations.indexOf(mutation);
+                    for (var i = 0; i <= mutationIndex; i++) {
+                        var e = $scope.gene.mutations.get(i);
```

Comment:
That is a very thoughtful suggestion. It is possible to process it in O(n) instead of O(n square). I will change that